app/models/manageiq/providers/vmware/infra_manager/vm/remote_console.rb

```diff
@@ -101,6 +86,16 @@ def remote_console_vnc_acquire_ticket(proxy_miq_server = nil)
     end
     update_attributes(:vnc_port => host_port)
 
-    return password, host_address, host_port, proxy_address, proxy_port
+    Console.where(:vm_id => id).each(&:destroy)
+    Console.create!(
```

Comment:
Do these 2 operations always come together?  If so, perhaps there should be a method on Console that does the job of both so the caller does not have to be aware of it.